api4/role.go

```diff
@@ -29,12 +29,28 @@ var notAllowedPermissions = []string{
 }
 
 func (api *API) InitRole() {
+	api.BaseRoutes.Roles.Handle("", api.ApiSessionRequiredTrustRequester(getAllRoles)).Methods("GET")
 	api.BaseRoutes.Roles.Handle("/{role_id:[A-Za-z0-9]+}", api.ApiSessionRequiredTrustRequester(getRole)).Methods("GET")
 	api.BaseRoutes.Roles.Handle("/name/{role_name:[a-z0-9_]+}", api.ApiSessionRequiredTrustRequester(getRoleByName)).Methods("GET")
 	api.BaseRoutes.Roles.Handle("/names", api.ApiSessionRequiredTrustRequester(getRolesByNames)).Methods("POST")
 	api.BaseRoutes.Roles.Handle("/{role_id:[A-Za-z0-9]+}/patch", api.ApiSessionRequired(patchRole)).Methods("PUT")
 }
 
+func getAllRoles(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.IsSystemAdmin() {
+		c.Err = model.NewAppError("getAllRoles", "api.role.get_all_roles.insufficient_permissions", nil, "", http.StatusForbidden)
```

Comment:
This string needs to be added to the translations (/i18n/en.json)

Reply:
@wiggin77 I think instead of using 

```go
if !c.IsSystemAdmin() {
   c.Err = model.NewAppError("getAllRoles", "api.role.get_all_roles.insufficient_permissions", nil, "", http.StatusForbidden)
}
```
We should use
```go
if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
  c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
  return
}
```

I didnt realize this was the way permissions errors were handled. This way it will be consistent with the other functions and i18n localizations are already there.